source/extensions/filters/udp/udp_proxy/udp_proxy_filter.cc

```diff
@@ -60,6 +65,40 @@ Network::FilterStatus UdpProxyFilter::onReceiveError(Api::IoError::IoErrorCode)
   return Network::FilterStatus::StopIteration;
 }
 
+UdpProxyFilter::~UdpProxyFilter() {
+  if (!config_->accessLogs().empty()) {
+    fillStreamInfo();
```

Comment:
Are these stats only updated in dtor? I am not sure if these stats should be updated every time when data come.

Reply:
And I also doubt the design writing access log only in dtor. In TCP, every connection has its own TCP proxy filter, but a UDP proxy filter works for different datagrams coming from different source.

Reply:
> And I also doubt the design writing access log only in dtor. In TCP, every connection has its own TCP proxy filter, but a UDP proxy filter works for different datagrams coming from different sources.



> Are these stats only updated in dtor? I am not sure if these stats should be updated every time when data come.

Yeah, stats in udp proxy are updated during the whole lifetime, including sending and receiving data.

Reply:
I think this is a legitimate question, as I'm not sure this is what the user will intend? This is going to rarely log as the filter will live for the lifetime of the listener. I think you probably actually want to log for each session?

Reply:
> I think this is a legitimate question, as I'm not sure this is what the user will intend? This is going to rarely log as the filter will live for the lifetime of the listener. I think you probably actually want to log for each session?

Yeah, I thought about this before, udp proxy uses shared stats for all sessions currently, but this may require each session to record its own stats.

Reply:
I don't think it's too difficult to maintain per session stats. The way I would probably implement it is to just create a StreamInfoImpl for each session, and put the needed "stats" in there via increments of whatever you need as the session moves along. If we don't do this I think it would be too confusing for the user?

/wait